gen/calc.py

```diff
@@ -172,7 +172,11 @@ def calculate_ip_detect_contents(ip_detect_filename):
     return yaml.dump(open(ip_detect_filename, encoding='utf-8').read())
 
 
-def calculate_ip_detect_public_contents(ip_detect_contents):
+def calculate_ip_detect_public_contents(ip_detect_contents, ip_detect_public_filename):
+    if ip_detect_public_filename != '':
```

Comment:
The function on line 170 does *not* do the same thing. These are calc functions and so their arguments become gen.internals.Target's . Thus, this allows you to provide 'ip_detect_public_filename' which was not the case before